From 3e485ded83ebede1e00fe4fb16bcade9c089d213 Mon Sep 17 00:00:00 2001
From: dkati <dkatikaridis@gmail.com>
Date: Wed, 27 Jun 2018 23:58:31 +0300
Subject: [PATCH 199/201] CameraSettings: Do not crash if zoom ratios are not
 exposed

* On some devices, wrappers do not expose the zoom ratios, so the
  list of zoomRatios is empty. So the .add() method cant run from a
  null class. If zoom rations are not exposed, simply return an empty
  list and the camera will take the default HAL's zoom ratios

-This fixes front camera photo shoot on jfltexx
Change-Id: Ic64e41a8c5a2d51966ec95d4b020e5d3951eac0e
---
 src/com/android/camera/CameraSettings.java | 9 +++++----
 1 file changed, 5 insertions(+), 4 deletions(-)

diff --git a/src/com/android/camera/CameraSettings.java b/src/com/android/camera/CameraSettings.java
index 2023f0fca..f32b12182 100755
--- a/src/com/android/camera/CameraSettings.java
+++ b/src/com/android/camera/CameraSettings.java
@@ -813,10 +813,11 @@ public class CameraSettings {
         ArrayList<String> supported = new ArrayList<String>();
         int zoomMaxIdx = params.getMaxZoom();
         List <Integer>  zoomRatios = params.getZoomRatios();
-        int zoomMax = zoomRatios.get(zoomMaxIdx)/100;
-
-        for (int zoomLevel = 0; zoomLevel <= zoomMax; zoomLevel++) {
-            supported.add(String.valueOf(zoomLevel));
+        if (zoomRatios != null ) {
+            int zoomMax = zoomRatios.get(zoomMaxIdx)/100;
+            for (int zoomLevel = 0; zoomLevel <= zoomMax; zoomLevel++) {
+                supported.add(String.valueOf(zoomLevel));
+            }
         }
         return supported;
     }
-- 
2.17.1

